### PR TITLE
add sonarqube

### PR DESCRIPTION
**Pull request by udernaeb**
This is a test of pull request.

### CODE REVIEW

**Comment by freefri** [comment]
@udernaeb I have add one more commit, including your command from the readme into the docker compose file (the reason to use docker compose, is to be able to add easily there any other docker service like sonarqube, wihtout running any other manual command.

Please review my change and update the readme file with some example on how to run the sonar checks from our development machines (we can talk about this on our meeting tomorrow)

Changing topic: PR looks great. I have pushed some new changes to master, to test what is happening with conflicts and how github is solving them (I am not familiar with github, but it should be pretty straight forward)

**Comment by udernaeb** [comment]
> @udernaeb I have add one more commit, including your command from the readme into the docker compose file (the reason to use docker compose, is to be able to add easily there any other docker service like sonarqube, wihtout running any other manual command.
> 
> Please review my change and update the readme file with some example on how to run the sonar checks from our development machines (we can talk about this on our meeting tomorrow)
> 
> Changing topic: PR looks great. I have pushed some new changes to master, to test what is happening with conflicts and how github is solving them (I am not familiar with github, but it should be pretty straight forward)

Better we check tomorrow, because I'm not 100% sure yet.

**Comment by freefri** [comment]
@udernaeb I have set up a pipeline to run code style and tests (take a look to https://github.com/oreplay/oreplay-backend/actions/runs/8142471391/job/22252103874 ) 

If sonarqube is working on your local, we could add it as well. We should schedule a 30 minute call to talk about what is missing and how to proceed.

**Comment by udernaeb** [comment]
> @udernaeb I have set up a pipeline to run code style and tests (take a look to https://github.com/oreplay/oreplay-backend/actions/runs/8142471391/job/22252103874 )
> 
> If sonarqube is working on your local, we could add it as well. We should schedule a 30 minute call to talk about what is missing and how to proceed.

I will take a look today, but we can take a look together.

**Comment by udernaeb** [comment]
Seems that we need to put the sonarqube code in main branch in order to execute it.

File: sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+sonar.projectKey=oreplay_oreplay-backend_39571abe-567d-4bd3-b863-d41f763959a4
```
Comment (freefri) [inline sonar-project.properties:1]: if this is a secret it should not be commited

**Review by freefri — APPROVED** [review]
approved

**Comment by freefri** [comment]
I have set up this repo with [rebase](https://git-scm.com/docs/git-rebase) instead of merge [because](https://www.atlassian.com/git/tutorials/merging-vs-rebasing) I think rebase makes a much cleaner history and easier to track problems.

The original sonarqube branch is very old and we have got conflicts (I think mainly in the readme file) for this reason, rebase in github web interface is not allowed.

I am rebasing manually using git on my local and submiting a new PR https://github.com/oreplay/oreplay-backend/pull/11

**Pull request closed without merging**